### DISCUSSION TITLE
Internationalisation of the content entered on paragraph block

### DISCUSSION THREAD

**Discussion by awaheedm** [codex-team/editor.js#2347] [Q&A]
Hi,

We are evaluating wysiwyg editors at the moment, Editor.js seem to be the most promising editor of all. However, one of our core requirements is to allow our users to provide multiple translations on paragraphs or image caption etc.. From the documentation we checked we find the ability to provide translations on labels/text of the editor, but it is not clear if there is any ability to provide multiple text elements on the paragraph block. We were looking for a model that could be some thing like (see below). Perhaps there is a way to achieve this, or it can only be done by adding custom block implementation?

Any pointer in the right direction is greatly appreciated.

```
"blocks": [
  {
    id: "mhTl6ghSkV",
    type: "paragraph",
    data: {
      text: [
        {
          "lang": "en",
          "value": "Text in english"
        },
        {
          "lang": "de",
          "value": "Text in German"
        }
      ]
    }
  }
]
```

**Comment by phtaylor** [comment]
I came here to ask the same question, but I might as well try to anser it.

Fork the default Paragraph block.

https://github.com/editor-js/paragraph/blob/master/src/index.ts#L50-L55
expand the ParagraphData with per langauge text as you decribed.

Add a configuration value to select the current active language. 

Modfy the code to select the text based on that language selection.

Make your new block the default block in your application...

https://github.com/codex-team/editor.js/blob/fdaef55e5fc54d4a48a2e6628a5399151a8ee008/types/configs/editor-config.d.ts#L29

We may end up building this as we need it also.

**Reply by phtaylor** [reply]
I started working on this here.

https://github.com/ZeaInc/paragraph-i18n/tree/add-i18n

**Reply by neSpecc** [reply]
> I started working on this here.
> 
> https://github.com/ZeaInc/paragraph-i18n/tree/add-i18n

Could you share how it looks for the end user at the editor? How to enter a text for different languages?
